CMakeLists.txt

```diff
@@ -1,3 +1,7 @@
+if("${PROJECT_SOURCE_DIR}" STREQUAL "${PROJECT_BINARY_DIR}")
```

Comment:
Is this a standard approach? I couldn't find any guildelines regarding this online.

Because it isn't explicitly disallowed by CMake, I don't know if we want to arbitrarily turn the feature off.

Also, everyone has to make a complete mess of their repo at least once to truly understand two key lessons in life:
1. you don't do in tree builds
2. that CMake can't be trusted
;)

Reply:
I can't conceive of any possible reason to make an in-source build, but I can think of several that cause me to accidentally do so, one of which prompted this PR today.